src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,293 @@
+import { ImportMock, MockManager } from 'ts-mock-imports';
+import { spy, restore } from 'sinon';
+import * as contractWrappersModule from '@polymathnetwork/contract-wrappers';
+import * as contextModule from '../../Context';
+import * as wrappersModule from '../../PolymathBase';
+import * as tokenFactoryModule from '../../testUtils/MockedTokenFactoryObject';
+import * as dividendFactoryModule from '~/entities/factories/DividendDistributionFactory';
+import * as pushDividendPaymentModule from '../../procedures/PushDividendPayment';
+import { PushDividendPayment } from '../../procedures/PushDividendPayment';
+import { Procedure } from '~/procedures/Procedure';
+import { ProcedureType, DividendType, ErrorCode } from '~/types';
+import { PolymathError } from '~/PolymathError';
+import { mockFactories } from '~/testUtils/mockFactories';
+import { Factories } from '../../Context';
+import { SecurityToken, DividendDistribution } from '~/entities';
+
+const params = {
+  symbol: 'TEST1',
+  dividendIndex: 0,
+  shareholderAddresses: ['0x01'],
```

Comment:
Fede, I just have one major comment here. 
Checking the test metrics- there seems to be something else in the logic that you can test (branch coverage 80%) 
Uncovered lines are 98, 118. What are these lines?
Line 98:
```
    if (shareholderAddresses) {
```
And Line 118: 
```
index < shareholderAddressChunks.length - 1
```
This code makes me dizzy too, so I'd suggest consulting Jere if things are not clear.

What I _think_ is missing here is that you need to test the path the procedure takes when the shareholderAddresses are empty. Well, to be more specific, in line 98 we want to make sure that the tests take both paths, 
`shareholderAddresses` and `!shareholderAddresses` . I think that this is related as well to 118. I'm a little lost in respect to what index and shareholderAddressChunks exactly do, but I have a strong feeling that the test metrics are complaining because you are not testing both paths here (I think the case where `undefined` is also missing from your current test suite)
```  
index < shareholderAddressChunks.length - 1
              ? undefined
              : createPushDividendPaymentResolver(
                  factories,
                  symbol,...
```